source/fundamentals/data-formats/serialization.txt

```diff
@@ -0,0 +1,134 @@
+.. _csharp-serialization:
+
+=============
+Serialization
+=============
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to use the {+driver-long+} to perform
+serialization. Serialization is the process of mapping a C# object to a BSON
+document for storage in MongoDB.
+
+.. tip:: Serialization
+
+   To learn more about serialization, see the :wikipedia:`Serialization </serialization>`
+   article on Wikipedia.
+
+Serializers
+-----------
+
+Serializers are classes that handle the translation of C# objects to and
+from BSON documents. Most serializers implement the ``IBsonSerializer``
+interface. The {+driver-short+} has many built-in serializers made to handle
+primitive types, collection types, and custom classes.
+
+For a full list of available serializers, see the
+`Serializers namespace API documentation <{+api-root+}/N_MongoDB_Bson_Serialization_Serializers.htm>`__.
+
+Serializer Registry
+-------------------
+
+The serializer registry contains all registered serializers that are available
+to your application. Before you can use a serializer, you must add it to the
+serializer registry, as shown in the following example:
+
+.. code-block:: csharp
+
+   BsonSerializer.RegisterSerializer(new CustomTypeSerializer()); 
+
+To access the serializer registry, use the ``SerializerRegistry`` property
+of the ``BsonSerializer`` class as follows:
+
+.. code-block:: csharp
+
+   var intSerializer = BsonSerializer.SerializerRegistry.GetSerializer<int>();
+
+.. important::
+
+   The serializer registry is a global registry. This means that you cannot use
+   multiple registries in a single application.
+
+Custom Serializers
+~~~~~~~~~~~~~~~~~~
+
+In some cases, you might need to create a custom serializer. When creating a
+custom serializer, implement the ``SerializerBase<T>`` abstract base class and
+override the ``Deserialize()`` and ``Serialize()`` methods.
+
+The following code example shows a custom ``BsonRegularExpression`` serializer:
+
+.. code-block:: csharp
+
+   class CustomRegularExpressionSerializer : SerializerBase<BsonRegularExpression>
```

Comment:
This should be `Regex` rather than `BsonRegularExpression`. It isn't common to use BSON types (other than maybe `BsonDocument`) directly in your C# application. The serializer's job is to convert from database types (like `BsonRegularExpression`) to C# types like `Regex`.